python/tank/context.py

```diff
@@ -668,6 +668,83 @@ def create_empty(tk):
     """
     return Context(tk)
 
+def from_entities(tk, project, entity, step, task, user):
+    """
+    Construct a new context from the specified entities.
```

Comment:
What if the links between these entities don't match? What if you specify project X and then entity A which links to project Y. The existing context concepts are all working within these constraints - should we be worried that this method doesn't?